exercises/practice/yacht/.meta/example.c

```diff
@@ -0,0 +1,108 @@
+#include "yacht.h"
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+static int number_of_x(dice_t dice, int x)
+{
+   int sum = 0;
+   for (int i = 0; i < 5; ++i) {
+      if (dice.faces[i] == x)
+         sum += x;
+   }
+   return sum;
+}
+
+typedef struct {
+   uint8_t faces[6];
```

Comment:
```suggestion
   int faces[6];
```

There's definitely nothing wrong with a fixed sized integer, but it's inconsistent with the typing of the rest of the solution. If we change this to `int` we should be able to remove `stdint.h` above.

Reply:
If you'd rather use fixed-width types consistently, that's also fine with me.

Reply:
Where exactly is this inconsistent? I use `uint8_t` everywhere except for the final score. Technically, that could also be a `uint8_t`, but some small change in the specification (let's say yacht is is not 50, but 300 points) would break this.

My main motivation here, though, was efficiency. 6 bytes will easily be passed via a register when doing function calls. 24 bytes could be a different story.

I can change it to `int` if you prefer. But I'd keep the `uint8_t` here.

Reply:
There are several places in this that use `int` instead of a fixed-width type. I understand your reasons for making the choice you made, but this is a learning aid, not production code. This optimization will have little meaning to someone scoring a dice game to learn the fundamentals of C. My thought is that seeing a mix of fixed-width and non-fixed-width types will be more surprising to someone solving the problem than a missed optimization.